articles/150_roslaunch.md

```diff
@@ -0,0 +1,170 @@
+---
+layout: default
+title: ROS 2 Launch System
+permalink: articles/roslaunch.html
+abstract:
+  The launch system in ROS is responsible for helping the user describe the configuration of their system and then execute it as described. The configuration of the system includes what programs to run, what arguments to pass them, and ROS specific conventions which make it easy to reuse components throughout the system through configuration. Also, because the launch system is the entity that executes processes, it is also responsible for monitoring the state of processes in the system, then reporting and/or reacting to changes in the state of those processes.
+author: '[William Woodall](https://github.com/wjwwood)'
+published: true
+---
+
+- This will become a table of contents (this text will be scraped).
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Context
+
+<div class="alert alert-warning" markdown="1">
+TODO: Fill out some context with respect to ROS 1's `roslaunch` and other similar tools which exist in OS's.
+</div>
+
+## Separation of Concern
+
+The launch system can be considered in parts, separated by concern.
+The coarse breakdown is like so:
+
+- Calling Conventions for Processes and Various Styles of Nodes
+- System Description
```

Comment:
That's a good question. That's what has been consuming my time when working on this document is finding the right partition between the value of a static format that can be edited with tools and introspected without running it, versus a fully dynamic system that can do anything, but also is unpredictable without executing it (something like the halting problem, but not quite). It could be addressed by having a limited, declarative style for the description but being implemented within a more expressive syntax than xml or yaml. There are whole projects for this topic, basically using something like Python or Ruby to make a DSL for a specific topic or domain.

I've collected a lot of information on this subject, but I'm still working out how to present it most clearly and come up with a suggestion on how to proceed before discussing it.

To summarize, I think it's clear we need dynamic behavior (as you said "ways to launch specific components following dynamic device discovery"), but the question for me is "does the static description need to support that?" or "should that use case be solved by using the launch system's API, completely outside of the static description?".